src/webgpu/gpu_test.ts

```diff
@@ -167,6 +167,33 @@ export class GPUTest extends Fixture {
     });
   }
 
+  expectAlternativeContents(
+    src: GPUBuffer,
+    expected1: TypedArrayBufferView,
+    expected2: TypedArrayBufferView,
+    srcOffset: number = 0
+  ): void {
+    assert(expected1.byteLength === expected2.byteLength);
+    const { dst, begin, end } = this.createAlignedCopyForMapRead(
+      src,
+      expected1.byteLength,
+      srcOffset
+    );
+
+    this.eventualAsyncExpectation(async niceStack => {
+      const constructor = expected1.constructor as TypedArrayBufferViewConstructor;
+      await dst.mapAsync(GPUMapMode.READ);
+      const actual = new constructor(dst.getMappedRange());
+      const check1 = this.checkBuffer(actual.subarray(begin, end), expected1);
+      const check2 = this.checkBuffer(actual.subarray(begin, end), expected2);
+      if (check1 !== undefined && check2 !== undefined) {
+        niceStack.message = check1 + check2;
```

Comment:
IIRC just concatenating these messages isn't going to make much sense. Maybe add an explanation (and a newline):
```suggestion
        niceStack.message = `Expected one of the following two checks to succeed:
- ${check1}
- ${check2}`;
```